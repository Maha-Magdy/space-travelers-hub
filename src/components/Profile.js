/* eslint-disable react/prop-types */
import React from 'react';
import { connect } from 'react-redux';

const Profile = (props) => {
  const { rockets } = props;
  const { missions } = props;
  console.log('Hello', props);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved === true);
  const reservedMissions = missions.filter((mission) => mission.reserved === true);
  return (
    <div className="container-fluid profile">
      <div className="row align-items-start">
        <div className="col">
          <h5>My Missions</h5>
          {reservedMissions.length === 0 && (
            <p>There&apos;s no reserved missions</p>
          )}
          <ul className="p-0 list-unstyled">
            {reservedMissions.map((reservedMission) => (
              <li key={reservedMission.mission_id} className="p-2">{reservedMission.mission_name}</li>
            ))}
          </ul>
        </div>
        <div className="col">
          <h5>My Rockets</h5>
          {reservedRockets.length === 0 && (
            <p>There&apos;s no reserved rockets</p>
          )}
          <ul className="p-0 list-unstyled">
            {reservedRockets.map((reservedRocket) => (
              <li key={reservedRocket.id} className="p-2">{reservedRocket.name}</li>
            ))}
          </ul>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => ({
  rockets: state.rocketsReducer,
  missions: state.missionsReducer,
});

export default connect(mapStateToProps)(Profile);
