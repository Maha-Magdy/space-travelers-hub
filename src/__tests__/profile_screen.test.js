import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Profile from '../components/Profile';

it('Profile screen renders as expected', () => {
  const tree = renderer.create(<Provider store={store}><Profile /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
