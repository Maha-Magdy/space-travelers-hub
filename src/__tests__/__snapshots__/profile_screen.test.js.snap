// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile screen renders as expected 1`] = `
<div
  className="container-fluid profile"
>
  <div
    className="row align-items-start"
  >
    <div
      className="col"
    >
      <h5>
        My Missions
      </h5>
      <p>
        There's no reserved missions
      </p>
      <ul
        className="p-0 list-unstyled"
      />
    </div>
    <div
      className="col"
    >
      <h5>
        My Rockets
      </h5>
      <p>
        There's no reserved rockets
      </p>
      <ul
        className="p-0 list-unstyled"
      />
    </div>
  </div>
</div>
`;
